tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,379 @@
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "123\n", "jkl\n", "f\n"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+    }
+
+    /**
+     * Get new instance of the ballerina server.
+     *
+     * @throws BallerinaTestException
+     */
+    private void getNewInstanceOfBallerinaServer() throws BallerinaTestException {
+        ballerinaServer = new ServerInstance(serverZipPath);
+    }
+
+    /**
+     * Run and test main function in project.
+     *
+     * @param projectPath path of the project
+     * @param pkg         package name or balx file path
+     * @throws BallerinaTestException
+     */
+    private void runMainFunction(Path projectPath, String pkg) throws BallerinaTestException {
+        getNewInstanceOfBallerinaServer();
+        String[] clientArgsForRun = {"--sourceroot", projectPath.toString(), pkg};
+        LogLeecher logLeecher = new LogLeecher("Hello World!");
+        ballerinaServer.addLogLeecher(logLeecher);
+        ballerinaServer.runMain(clientArgsForRun, getEnvVariables(), "run");
+    }
+
+    /**
+     * Run and test service in project.
+     *
+     * @param serviceBalPath path of the service bal file
+     * @throws BallerinaTestException
+     * @throws IOException
+     */
+    private void runService(Path serviceBalPath) throws BallerinaTestException, IOException {
+        ServerInstance ballerinaServerForService = ServerInstance.initBallerinaServer();
+        ballerinaServerForService.startBallerinaServer(serviceBalPath.toString());
+        HttpResponse response = HttpClientRequest.doGet(ballerinaServerForService.getServiceURLHttp("hello/sayHello"));
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        ballerinaServerForService.stopServer();
+    }
+
+    /**
+     * Get environment variables and add ballerina_home as a env variable the tmp directory.
+     *
+     * @return env directory variable array
+     */
+    private String[] getEnvVariables() {
+        List<String> variables = new ArrayList<>();
+
+        Map<String, String> envVarMap = System.getenv();
+        envVarMap.forEach((key, value) -> variables.add(key + "=" + value));
+        return variables.toArray(new String[variables.size()]);
+    }
+
+    @AfterClass
+    private void cleanup() throws Exception {
+        deleteFiles(tempProjectDirectory);
+    }
+
+    /**
+     * Delete files inside directories.
+     *
+     * @param dirPath direectory path
+     * @throws IOException throw an exception if an issue occurs
+     */
+    private void deleteFiles(Path dirPath) throws IOException {
+        Files.walk(dirPath)
```

Comment:
Does the file walker work in windows ?

Reply:
Yes it works in windows

Reply:
Does it delete the files ?

Reply:
Yes it should delete the files. Since we are using Files.delete [1] it will throw an exception if the deletion fails

[1] https://docs.oracle.com/javase/7/docs/api/java/io/File.html#delete()

Reply:
In windows theres an issues where files created by java cannot be deleted due to a permission issue. Since we already tested it on windows, we can proceed.